nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/lazy/AbstractLazyCachableRenditionProvider.java

```diff
@@ -56,48 +56,46 @@
 
     public static final String CACHE_NAME = "LazyRenditionCache";
 
-    protected static Log log = LogFactory.getLog(AbstractLazyCachableRenditionProvider.class);
+    protected static Logger log = LogManager.getLogger(AbstractLazyCachableRenditionProvider.class);
 
     @Override
     public List<Blob> render(DocumentModel doc, RenditionDefinition definition) {
-        if (log.isDebugEnabled()) {
-            log.debug(String.format("Asking \"%s\" rendition lazy rendering for document %s (id=%s).",
-                    definition.getName(), doc.getPathAsString(), doc.getId()));
-        }
+        log.debug("Asking \"{}\" rendition lazy rendering for document {} (id={}).", definition::getName,
+                doc::getPathAsString, doc::getId);
 
         // Build the rendition key and get the current source document modification date
         String key = buildRenditionKey(doc, definition);
         String sourceDocumentModificationDate = getSourceDocumentModificationDate(doc, definition);
 
         // If rendition is not already in progress schedule it
-        List<Blob> blobs = null;
+        List<Blob> blobs;
         TransientStore ts = getTransientStore();
         if (!ts.exists(key)) {
             blobs = handleNewRendition(key, doc, definition, sourceDocumentModificationDate);
         } else {
-            String storedSourceDocumentModificationDate = (String) ts.getParameter(key,
+            String tsSourceDocumentModificationDate = (String) ts.getParameter(key,
                     SOURCE_DOCUMENT_MODIFICATION_DATE_KEY);
             blobs = ts.getBlobs(key);
             if (ts.isCompleted(key)) {
                 handleCompletedRendition(key, doc, definition, sourceDocumentModificationDate,
-                        storedSourceDocumentModificationDate, blobs);
+                        tsSourceDocumentModificationDate, blobs);
             } else {
                 handleIncompleteRendition(key, doc, definition, sourceDocumentModificationDate,
-                        storedSourceDocumentModificationDate);
+                        tsSourceDocumentModificationDate);
             }
         }
+        final List<Blob> finalBlobs = blobs;
+        log.debug("Returning blobs: {}.", () -> getBlobInfo(finalBlobs));
+        return blobs;
+    }
 
-        if (log.isDebugEnabled()) {
-            String blobInfo = null;
-            if (blobs != null) {
-                blobInfo = blobs.stream()
-                                .map(blob -> String.format("{filename=%s, MIME type=%s}", blob.getFilename(),
-                                        blob.getMimeType()))
-                                .collect(Collectors.joining(",", "[", "]"));
-            }
-            log.debug(String.format("Returning blobs: %s.", blobInfo));
+    protected String getBlobInfo(List<Blob> blobs) {
```

Comment:
usually i add  `@since` even for protected methods

Reply:
Really an internal method for logging purpose, but sure, it's always clearer.